libraries/abstractions/common_io/test/test_iot_flash.c

```diff
@@ -340,7 +340,6 @@ TEST( TEST_IOT_FLASH, AFQP_IotFlashReadInfo )
         TEST_ASSERT_EQUAL( 0, pxFlashInfo->ulPageSize & ( pxFlashInfo->ulPageSize - 1 ) );
         TEST_ASSERT_EQUAL( 0, pxFlashInfo->ulSectorSize & ( pxFlashInfo->ulSectorSize - 1 ) );
         TEST_ASSERT_EQUAL( 0, pxFlashInfo->ulBlockSize & ( pxFlashInfo->ulBlockSize - 1 ) );
-        TEST_ASSERT_EQUAL( 0, pxFlashInfo->ulFlashSize & ( pxFlashInfo->ulFlashSize - 1 ) );
```

Comment:
Can you provide some background on removing this?

Reply:
Glad you brought this up. I'm second guessing this change.

The idea was that `pxFlashInfo->ulFlashSize` is limited to _useable_ flash. For this board, our bootloader is stored at the end of total flash region, which I deemed _unuseuable_, thereby excluding some pages from the total of 1 MB (2^20) flash.

Technically, it's also not required that any of these are powers of 2. Though practically, it's a de-facto standard.
I'm thinking I'll re-introduce this condition and adjust value of `ulFlashSize` to include bootloader flash region. Thoughts?

Reply:
That sounds reasonable to me, reading the API it doesn't seem to differentiate. I'm unfamiliar with the Nordic APIs, but alternatively the different flash sections could be moved to different handles, so the bootloader can be managed separately. Curious what you think about this, seems the API (to me) is flexible in this regard, but will have a better idea regarding this

Reply:
Added this assertion in latest commit